utils/ffmpegvideodecode/ffmpeg_video_dec.cpp

```diff
@@ -194,19 +194,27 @@ int FFMpegVideoDecoder::HandleVideoSequence(RocdecVideoFormat *p_video_format) {
         if (!dec_context_) {
             THROW("Could not allocate video codec context");
         }
+        // set codec to automatically determine how many threads suits best for the decoding job
+        dec_context_->thread_count = 0;
+
+        if (decoder_->capabilities & AV_CODEC_CAP_FRAME_THREADS)
+            dec_context_->thread_type = FF_THREAD_FRAME;
+        else if (decoder_->capabilities & AV_CODEC_CAP_SLICE_THREADS)
+            dec_context_->thread_type = FF_THREAD_SLICE;
+        else
+            dec_context_->thread_count = 1;
+
         // open the codec
         if (avcodec_open2(dec_context_, decoder_, NULL) < 0) {
             THROW("Could not open codec");
         }
         // get the output pixel format from dec_context_
         decoder_pixel_format_ = (dec_context_->pix_fmt == AV_PIX_FMT_NONE) ? AV_PIX_FMT_YUV420P : dec_context_->pix_fmt;
-        dec_context_->thread_count = 4;
-        dec_context_->thread_type = FF_THREAD_FRAME;
-
+        //std::cout << "AVCodec delay: " << dec_context_->delay << "num B frames: " << dec_context_->max_b_frames<< std::endl;
     }
-    // allocate av_frame buffer pool for number of surfaces
+    // allocate av_frame buffer pool for number of surfaces to be in the decoder pool (we need delay + 4 to account for B frames)
```

Comment:
Please add a note saying that the number 4 is arbitrary. A proper calculation of the required delay will be provided later.